packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts

```diff
@@ -702,32 +695,47 @@ export class GraphqlApi extends GraphqlApiBase {
   }
 
   private setupSourceApiAssociations() {
-    this.definition.sourceApiOptions?.sourceApis.forEach(sourceApiOption => {
-      new CfnSourceApiAssociation(this, `${sourceApiOption.sourceApi.node.id}Association`, {
-        sourceApiIdentifier: sourceApiOption.sourceApi.apiId,
-        mergedApiIdentifier: this.api.attrApiId,
+    this.definition.sourceApiOptions?.sourceApis.forEach(sourceApiConfig => {
+      const mergeType = sourceApiConfig.mergeType ?? MergeType.AUTO_MERGE;
+      var sourceApiIdentifier = sourceApiConfig.sourceApi.apiId;
+      var mergedApiIdentifier = this.apiId;
+
+      // This is protected by a feature flag because if there is an existing source api association that used the api id,
+      // updating it to use ARN as identifier leads to a resource replacement. ARN is recommended going forward because it allows support
+      // for both same account and cross account use cases.
+      if (FeatureFlags.of(this).isEnabled(cxapi.APPSYNC_ENABLE_USE_ARN_IDENTIFIER_SOURCE_API_ASSOCIATION)) {
+        sourceApiIdentifier = sourceApiConfig.sourceApi.arn;
+        mergedApiIdentifier = this.arn;
+      }
+
+      const association = new CfnSourceApiAssociation(this, `${sourceApiConfig.sourceApi.node.id}Association`, {
+        sourceApiIdentifier: sourceApiIdentifier,
+        mergedApiIdentifier: mergedApiIdentifier,
         sourceApiAssociationConfig: {
-          mergeType: sourceApiOption.mergeType ?? MergeType.AUTO_MERGE,
+          mergeType: mergeType,
         },
+        description: sourceApiConfig.description,
       });
+
+      // Add permissions to merged api execution role, only if it was not passed in.
+      if (!this.definition.sourceApiOptions?.mergedApiExecutionRole) {
+        const executionRole = this.mergedApiExecutionRole as IRole;
```

Comment:
This is not the right way to allow that use case.

If people are passing in a role and they don't want it modified, they can pass [role.withoutPolicyUpdates()](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.Role.html#withoutwbrpolicywbrupdatesoptions), or import an existing one with `Role.fromRoleName({mutable: false })`.

Reply:
The existing support works like this where it is not updating the role if it was passed in. Would it be a breaking change?

Reply:
it might be... since this is pre-existing we can look at changing this separately.